x-pack/plugins/ml/public/alerting/preview_alert_condition.tsx

```diff
@@ -0,0 +1,277 @@
+
+  const validators = useMemo(
+    () => composeValidators(requiredValidator(), timeIntervalInputValidator()),
+    []
+  );
+
+  const validationErrors = useMemo(() => validators(lookBehindInterval), [lookBehindInterval]);
+
+  useEffect(
+    function resetPreview() {
+      setPreviewResponse(undefined);
+    },
+    [alertParams]
+  );
+
+  const testCondition = useCallback(async () => {
+    try {
+      const response = await alertingApiService.preview({
+        alertParams,
+        timeRange: lookBehindInterval!,
+      });
+      setPreviewResponse(response);
+      setPreviewError(undefined);
+    } catch (e) {
+      setPreviewResponse(undefined);
+      setPreviewError(e.body ?? e);
+    }
+  }, [alertParams, lookBehindInterval]);
+
+  /**
+   * Sample of the 5 results from the top hits
+   */
+  const sampleHits = useMemo(() => {
+    if (!previewResponse) return;
+
+    return previewResponse.results.slice(0, 5);
+  }, [previewResponse]);
+
+  const isReady =
+    (alertParams.jobSelection?.jobIds?.length! > 0 ||
+      alertParams.jobSelection?.groupIds?.length! > 0) &&
+    !!alertParams.resultType &&
+    !!alertParams.severity &&
+    validationErrors === null;
+
+  const isInvalid = lookBehindInterval !== undefined && !!validationErrors;
+
+  return (
+    <>
+      <EuiFlexGroup gutterSize="s" alignItems={'flexEnd'}>
+        <EuiFlexItem>
+          <EuiFormRow
+            label={
+              <FormattedMessage
+                id="xpack.ml.previewAlert.intervalLabel"
+                defaultMessage="Check the alert condition with an interval"
+              />
+            }
+            isInvalid={isInvalid}
+            error={invalidTimeIntervalMessage(lookBehindInterval)}
+          >
+            <EuiFieldText
+              placeholder="15d, 6m"
+              value={lookBehindInterval}
+              onChange={(e) => {
+                setLookBehindInterval(e.target.value);
+              }}
+              isInvalid={isInvalid}
+              data-test-subj={'mlAnomalyAlertPreviewInterval'}
+            />
+          </EuiFormRow>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiButton
+            onClick={testCondition}
+            disabled={!isReady}
+            data-test-subj={'mlAnomalyAlertPreviewButton'}
+          >
+            <FormattedMessage id="xpack.ml.previewAlert.testButtonLabel" defaultMessage="Test" />
+          </EuiButton>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+
+      {previewError !== undefined && (
+        <>
+          <EuiSpacer size="m" />
+          <EuiCallOut
+            title={
+              <FormattedMessage
+                id="xpack.ml.previewAlert.previewErrorTitle"
+                defaultMessage="Unable to load the preview"
+              />
+            }
+            color="danger"
+            iconType="alert"
+          >
+            <p>{previewError.message}</p>
```

Comment:
I am hitting an error when trying to preview an alert for a categorization job.

```
{"statusCode":400,"error":"Bad Request","message":"search_phase_execution_exception","attributes":{"body":{"error":{"root_cause":[{"type":"script_exception","reason":"runtime error","script_stack":["org.elasticsearch.index.fielddata.ScriptDocValues$Strings.get(ScriptDocValues.java:483)","org.elasticsearch.index.fielddata.ScriptDocValues$Strings.getValue(ScriptDocValues.java:497)","doc[\"timestamp\"].value + \"_\" + doc[\"function\"].value + \"_\" + doc[\"field_name\"].value","                                                                              ^---- HERE"],"script":"doc[\"timestamp\"].value + \"_\" + doc[\"function\"].value + \"_\" + doc[\"field_name\"].value","lang":"painless","position":{"offset":78,"start":0,"end":84}}],"type":"search_phase_execution_exception","reason":"all shards failed","phase":"query","grouped":true,"failed_shards":[{"shard":0,"index":".ml-anomalies-shared","node":"Odn2_DAXTe-WHhonpo8-3Q","reason":{"type":"script_exception","reason":"runtime error","script_stack":["org.elasticsearch.index.fielddata.ScriptDocValues$Strings.get(ScriptDocValues.java:483)","org.elasticsearch.index.fielddata.ScriptDocValues$Strings.getValue(ScriptDocValues.java:497)","doc[\"timestamp\"].value + \"_\" + doc[\"function\"].value + \"_\" + doc[\"field_name\"].value","                                                                              ^---- HERE"],"script":"doc[\"timestamp\"].value + \"_\" + doc[\"function\"].value + \"_\" + doc[\"field_name\"].value","lang":"painless","position":{"offset":78,"start":0,"end":84},"caused_by":{"type":"illegal_state_exception","reason":"A document doesn't have a value for a field! Use doc[<field>].size()==0 to check if a document is missing a field!"}}}]},"status":400}}}
```

Job config, using the Kibana sample log data is:
```
"analysis_config": {
    "bucket_span": "15m",
    "categorization_field_name": "message",
    "per_partition_categorization": {
      "enabled": false
    },
    "detectors": [
      {
        "detector_description": "count by mlcategory",
        "function": "count",
        "by_field_name": "mlcategory",
        "detector_index": 0
      }
    ],
    "influencers": [
      "mlcategory"
    ]
  },
```

Reply:
Fixed in d9724437d01ff51d953f3e1243480fb246c28374